cmd/promtool/main.go

```diff
@@ -364,6 +382,55 @@ func (ls lintConfig) lintDuplicateRules() bool {
 	return ls.all || ls.duplicateRules
 }
 
+// Check server health.
+func CheckServerHealth(url string) error {
+	if url == "" {
+		url = "http://localhost:9090"
+	}
+
+	// Check Health
+	healthCheckResp, err := http.Get(url + "/-/healthy")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "  FAILED:", err)
+		return err
+	} else if healthCheckResp.StatusCode != 200 {
+		return fmt.Errorf("health check failed: URL=%s, status=%d", url, healthCheckResp.StatusCode)
+	}
```

Comment:
The `if` condition is put on the `err` and if-else is put on `status` so these two are essentially two separate conditions that we are checking hence two separate if blocks make more sense here instead of else-if. Also, since the `if` block has a return statement in it, this else-if condition can easily be replaced by a normal `if` block without disturbing the logic.